src/features/postsSlice.ts

```
@@ -0,0 +1,53 @@
+/* eslint-disable no-param-reassign */
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { Post } from '../types/Post';
+import { getUserPosts } from '../api/posts';
+
+type PostState = {
+  items: Post[];
+  loaded: boolean;
+  hasError: boolean;
+};
+
+const initialState: PostState = {
+  items: [],
+  loaded: false,
+  hasError: false,
+};
+
+export const loadUserPosts = createAsyncThunk(
+  'posts/fetch',
+  async (id: number, { rejectWithValue }) => {
+    try {
+      const posts = await getUserPosts(id);
+
+      return posts;
+    } catch (error) {
+      return rejectWithValue(error);
+    }
+  },
+);
+
+export const postsSlice = createSlice({
+  name: 'posts',
+  initialState,
+  reducers: {},
+  extraReducers: builder => {
+    builder
+      .addCase(loadUserPosts.pending, state => {
+        state.loaded = false;
+        state.hasError = false;
+      })
+      .addCase(loadUserPosts.fulfilled, (state, action) => {
+        state.items = action.payload;
+        state.loaded = true;
+        state.hasError = false;
+      })
+      .addCase(loadUserPosts.rejected, state => {
```

Behavioral note (already mentioned in other reviews): on rejection you set `loaded = false` and `hasError = true`. That is fine, but be sure the UI shows a loader only while loading and not when `hasError` is true (`!loaded && !hasError` for loader). Otherwise the loader can mask the error message. The UI files (App.tsx and PostDetails.tsx) should be adjusted accordingly if necessary.